app/init.php

```diff
@@ -1629,6 +1631,76 @@ function getDevice(string $root, string $connection = ''): Device
     return $register->get('geodb');
 }, ['register']);
 
+App::setResource('ip', function ($request) {
+    return $request->getIp();
+}, ['request']);
+
+App::setResource('geodbClient', function () {
+    $client = new Client();
+    $client->addHeader('Authorization', 'Bearer ' . System::getEnv('_APP_GEO_SECRET'));
+    $client->setBaseUrl('http://appwrite-geo/v1');
+    return $client;
+});
+
+App::setResource('geoRecordDocker', function (string $ip, Client $geodbClient) {
+    $record = null;
+    try {
+        $record = $geodbClient->fetch("/ips/{$ip}", Client::METHOD_GET);
+        $record = $record->json();
+    } catch (Throwable $th) {
+        Console::error($th->getMessage());
+        Console::error($th->getTraceAsString());
+        $record = null;
+    }
+
+    return $record;
+}, ['ip', 'geodbClient']);
+
+App::setResource('geoRecordEmbeded', function (string $ip, Reader $geodb) {
```

Comment:
I'm a bit confused between the roles of the different objects. I'm pretty sure everything can be much simpler.